nio/src/main/scala/zio/nio/channels/AsynchronousChannel.scala

```diff
@@ -151,10 +139,7 @@ object AsynchronousByteChannel {
   private[channels] def effectAsyncChannel[C <: JChannel, A](
     channel: C
   )(op: C => CompletionHandler[A, Any] => Any): IO[IOException, A] =
-    IO.effectAsyncInterrupt { k =>
-      op(channel)(completionHandlerCallback(k))
-      Left(IO.effect(channel.close()).ignore)
-    }
+    Task(op(channel)).flatMap(Task.effectAsyncWithCompletionHandler).refineToOrDie[IOException]
```

Comment:
Unfortunately, I don't think we can do it like this because there's no way to interrupt `op(channel)`.